src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -2012,10 +2017,13 @@ bool RTPSParticipantImpl::register_in_reader(
         res = true;
         for ( auto reader : m_userReaderList)
         {
-            res &= reader->add_statistics_listener(listener);
+            if (!fastdds::statistics::is_statistics_builtin(reader->getGuid().entityId))
+            {
+                res &= reader->add_statistics_listener(listener);
+            }
         }
     }
-    else
+    else if (!fastdds::statistics::is_statistics_builtin(reader_guid.entityId))
```

Comment:
Statistics builtin endpoints will never be readers.

Reply:
just in case 😂